consensus/participant.go

```diff
@@ -136,24 +117,34 @@ func NewParticipant(
 		return nil, fmt.Errorf("could not initialize event loop: %w", err)
 	}
 
+	modules.QCCreatedDistributor.AddConsumer(loop.SubmitTrustedQC)
+
 	return loop, nil
 }
 
-func initForks(final *flow.Header, headers storage.Headers, updater module.Finalizer, notifier hotstuff.Consumer, rootHeader *flow.Header, rootQC *flow.QuorumCertificate) (*forks.Forks, error) {
-	finalizer, err := initFinalizer(final, headers, updater, notifier, rootHeader, rootQC)
+func InitForks(final *flow.Header, headers storage.Headers, updater module.Finalizer, modules *HotstuffModules, rootHeader *flow.Header, rootQC *flow.QuorumCertificate) (*HotstuffModules, error) {
+	finalizer, err := initFinalizer(final, headers, updater, modules.Notifier, rootHeader, rootQC)
 	if err != nil {
 		return nil, fmt.Errorf("could not initialize finalizer: %w", err)
 	}
 
 	// initialize the fork choice
-	forkchoice, err := forkchoice.NewNewestForkChoice(finalizer, notifier)
+	forkchoice, err := forkchoice.NewNewestForkChoice(finalizer, modules.Notifier)
 	if err != nil {
 		return nil, fmt.Errorf("could not initialize fork choice: %w", err)
 	}
 
-	// initialize the forks manager
-	forks := forks.New(finalizer, forkchoice)
-	return forks, nil
+	// initialize the Forks manager
+	modules.Forks = forks.New(finalizer, forkchoice)
+
+	return modules, nil
+}
+
+func InitValidator(metrics module.HotstuffMetrics, modules *HotstuffModules) *HotstuffModules {
+	// initialize the Validator
+	modules.Validator = validatorImpl.New(modules.Committee, modules.Forks, modules.Signer)
+	modules.Validator = validatorImpl.NewMetricsWrapper(modules.Validator, metrics) // wrapper for measuring time spent in Validator component
+	return modules
 }
```

Comment:
better to avoid mutation
```suggestion
func InitValidator(metrics module.HotstuffMetrics, modules *HotstuffModules) hotstuff.Validator {
	// initialize the Validator
	validator := validatorImpl.New(modules.Committee, modules.Forks, modules.Signer)
	return validatorImpl.NewMetricsWrapper(validator, metrics) // wrapper for measuring time spent in Validator component
}
```